src/main/blackbox/blackbox.c

```diff
@@ -353,8 +353,20 @@ static const blackboxSimpleFieldDefinition_t blackboxSlowFields[] = {
     {"wind",                   0, SIGNED,   PREDICT(0),      ENCODING(UNSIGNED_VB)},
     {"wind",                   1, SIGNED,   PREDICT(0),      ENCODING(UNSIGNED_VB)},
     {"wind",                   2, SIGNED,   PREDICT(0),      ENCODING(UNSIGNED_VB)},
-    {"temperature",           -1, SIGNED,   PREDICT(0),      ENCODING(SIGNED_VB)},
-    {"temperatureSource",     -1, UNSIGNED, PREDICT(0),      ENCODING(UNSIGNED_VB)},
+    {"IMUTemperature",        -1, SIGNED,   PREDICT(0),      ENCODING(SIGNED_VB)},
+#ifdef USE_BARO
+    {"baroTemperature",       -1, SIGNED,   PREDICT(0),      ENCODING(SIGNED_VB)},
+#endif
+#ifdef USE_TEMPERATURE_SENSOR
+    {"sens0Temp",             -1, SIGNED,   PREDICT(0),      ENCODING(SIGNED_VB)},
+    {"sens1Temp",             -1, SIGNED,   PREDICT(0),      ENCODING(SIGNED_VB)},
+    {"sens2Temp",             -1, SIGNED,   PREDICT(0),      ENCODING(SIGNED_VB)},
+    {"sens3Temp",             -1, SIGNED,   PREDICT(0),      ENCODING(SIGNED_VB)},
+    {"sens4Temp",             -1, SIGNED,   PREDICT(0),      ENCODING(SIGNED_VB)},
+    {"sens5Temp",             -1, SIGNED,   PREDICT(0),      ENCODING(SIGNED_VB)},
+    {"sens6Temp",             -1, SIGNED,   PREDICT(0),      ENCODING(SIGNED_VB)},
+    {"sens7Temp",             -1, SIGNED,   PREDICT(0),      ENCODING(SIGNED_VB)},
```

Comment:
> 
> 
> 8 temp sensors are probably an overkill. In the best case we have Gyro + CPU + baro + external. 4 slots should be enough.

@digitalentity When you stated the above. Did you mean only having one external sensor type to choose from?
I understand why @shellixyz  has given us this many choices for external sensors. 
Due to the difference in the sizes of each sensor. 
The smaller TO-92 case DS18 can be placed in the base of a motor, VTX or ESC. While the larger breakout board mount SO8 - LM75, can be mounted on a battery pack etc.

I believe versatility can be helpful : )

Reply:
I can lower the default max number of sensors but I think 5-6 on top of the IMU and baro ones  isn't unreasonable. For example somebody could monitor: battery, ESC, motor(s), air, VTX. The IMU and baro temperatures are most of the time only measuring the FC temp, not that useful. Also for BB it is not really an issue since temperatures are only recorded in slow frames and only active sensors will take bandwidth since they only would produce changing values.

Reply:
Makes sense. However, I think we can move temperature to slow blackbox frames to save bandwidth.

Reply:
They are recorded in slow frames

Reply:
Didn't notice 😄